### PR TITLE
removed dependency on junit's listener and used only surefire's repor…

### PR DESCRIPTION
**Pull request by alexgyori**
…ting to find test failures

### CODE REVIEW

File: nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/CleanSurefireExecution.java [hunk 1]
```diff
@@ -79,63 +87,67 @@ public void run() throws MojoExecutionException {
...
             Logger.getGlobal().log(Level.INFO, "Surefire failed when running tests for " + this.configuration.executionId);
+
+            SurefireReportParser parser = new SurefireReportParser(
+                    Arrays.asList(this.configuration.getExecutionDir().toFile()), Locale.getDefault());
+            try {
+                Set<String> failingTests = new LinkedHashSet<>();
+                for (ReportTestSuite report : parser.parseXMLReportFiles()) {
+                    for (ReportTestCase testCase : report.getTestCases()) {
+                        if (testCase.hasFailure()) {
+                            String testName = testCase.getFullName();
```
Comment (august782) [inline nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/CleanSurefireExecution.java]: Is there a way to get the separate test class and test method names from the ReportTestCase?

File: nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/CleanSurefireExecution.java [hunk 2]
```diff
@@ -79,63 +87,67 @@ public void run() throws MojoExecutionException {
...
 
-    private Xpp3Dom createChildIfNotExists(Xpp3Dom configuration, String child) {
-        Xpp3Dom childNode = configuration.getChild(child);
-        if (childNode == null) {
-            childNode = new Xpp3Dom(child);
-            configuration.addChild(childNode);
-        }
-        return childNode;
+    private Xpp3Dom makeNode(String nodeName, String value) {
+        Xpp3Dom outputDir = new Xpp3Dom(nodeName);
```
Comment (august782) [inline nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/CleanSurefireExecution.java]: Rename outputDir

**Comment by alexgyori** [comment]
@august782 Please check my last commit; I added count incrementing in the extendZoneStrings, since it seems it was missing (on top of the refactorings that are just beautyfying the name and making them more readable for when stopping in the debugger). I also addressed the prior comments you had. LGTM?

**Comment by alexgyori** [comment]
Fixes #11

**Pull request merged**